src/video/win32_v.cpp

```diff
@@ -1316,3 +1316,45 @@ Dimension VideoDriver_Win32::GetScreenSize() const
 {
 	return { static_cast<uint>(GetSystemMetrics(SM_CXSCREEN)), static_cast<uint>(GetSystemMetrics(SM_CYSCREEN)) };
 }
+
+ZoomLevel VideoDriver_Win32::GetSuggestedUIZoom()
+{
+	typedef UINT (WINAPI *PFNGETDPIFORWINDOW)(HWND hwnd);
+	typedef UINT (WINAPI *PFNGETDPIFORSYSTEM)(VOID);
+	typedef HRESULT (WINAPI *PFNGETDPIFORMONITOR)(HMONITOR hMonitor, int dpiType, UINT *dpiX, UINT *dpiY);
+
+	static PFNGETDPIFORWINDOW _GetDpiForWindow = nullptr;
+	static PFNGETDPIFORSYSTEM _GetDpiForSystem = nullptr;
+	static PFNGETDPIFORMONITOR _GetDpiForMonitor = nullptr;
+
+	static bool init_done = false;
+	if (!init_done) {
+		init_done = true;
+
+		_GetDpiForWindow = (PFNGETDPIFORWINDOW)GetProcAddress(GetModuleHandle(_T("User32")), "GetDpiForWindow");
+		_GetDpiForSystem = (PFNGETDPIFORSYSTEM)GetProcAddress(GetModuleHandle(_T("User32")), "GetDpiForSystem");
+		_GetDpiForMonitor = (PFNGETDPIFORMONITOR)GetProcAddress(LoadLibrary(_T("Shcore.dll")), "GetDpiForMonitor");
+	}
+
+	const UINT BASE_DPI = 96; // Default Windows DPI value.
```

Comment:
Maybe either move this out of the function or closer to where it is used? There is a bit of a distance now, and it is in the middle of a function. Was a bit hard to spot.